### DISCUSSION TITLE
New GitHub Autograding Action that reads and grades build results

### DISCUSSION THREAD

**Discussion by uhafner** [community/community#78157] [GitHub Education]
### Hi everyone,

I would like to announce a new [GitHub Autograding Action](https://github.com/marketplace/actions/autograding-action) (MIT license) that can be used as an alternative to the existing [Education Autograding Action](https://github.com/education/autograding). It uses a different concept: the action reads and interprets the build results of a GitHub Actions workflow run. This allows you to use any programming language, testing framework, or static analysis tool you want. The action can read, interpret, and grade unit test results, code coverage reports and static analysis reports. It reports the overall results in the pull request and additionally shows warnings and lines with missing coverage or survived mutations directly in the diff view of the pull request.

I use this action for student assignments in my software development and engineering courses at the Munich University of Applied Sciences. Hopefully, it is useful for others as well.

<img width="732" alt="pr-comment" src="https://github.com/community/community/assets/503338/ac79ec44-bf30-4243-898c-6fc06f1c1a79">

**Comment by pm-gmr** [comment]
How can we integrate it with gcc? I can see it supports a large number of report types, but I have no idea how to provide the report to it.

**Reply by uhafner** [reply]
You need to write or pipe the compiler output to a file (SARIF is preferred) and use that file as an input (in my example workflow I am  using `tee`: https://github.com/jenkinsci/analysis-model/blob/5ea175baf465edbd376b38ef8b713c389721724d/.github/workflows/quality-monitor.yml#L28). You then can use a pattern like:

```
"tools": [
                  {
                    "id": "gcc",
                    "name": "GCC Warnings",
                    "pattern": "**/compiler-log.txt"
                  }
                ],
```
